app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordScreen.kt

```diff
@@ -0,0 +1,307 @@
+                                    R.string.error_validation_blank,
+                                    passwordType,
+                                )
+                            }
+
+                            viewModel.isInputLengthInadequate(password) -> {
+                                passwordError = context.getString(
+                                    R.string.error_validation_minimum_chars,
+                                    passwordType,
+                                    context.resources.getInteger(R.integer.password_minimum_length),
+                                )
+                            }
+                        }
+                        return passwordError
+                    }
+
+                    newPasswordErrorContent = getPasswordError(newPassword.text, context.getString(R.string.new_password))
+                    confirmPasswordErrorContent = getPasswordError(confirmPassword.text, context.getString(R.string.confirm_password))
+                    passwordMatches = viewModel.validatePasswordMatch(newPassword.text, confirmPassword.text)
+
+                    when {
+                        newPasswordErrorContent.isEmpty() && confirmPasswordErrorContent.isEmpty() -> {
+                            if (passwordMatches && Network.isConnected(context)) {
+                                viewModel.updateAccountPassword(
+                                    newPassword = newPassword.text  ,
+                                    confirmPassword = confirmPassword.text
+                                )
+                            } else errorComponent(context)
+                        }
+
+                        newPasswordErrorContent.isEmpty() && confirmPasswordErrorContent.isNotEmpty() -> {
+                            confirmPasswordError = true
+                        }
+
+                        newPasswordErrorContent.isNotEmpty() && confirmPasswordErrorContent.isEmpty() -> {
+                            newPasswordError = true
+                        }
+
+                        else -> {
+                            confirmPasswordError = true
+                            newPasswordError = true
+                        }
+                    }
+                },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 16.dp, end = 16.dp, top = 4.dp),
+                contentPadding = PaddingValues(12.dp),
+                colors = ButtonDefaults.buttonColors(
+                    containerColor = MaterialTheme.colorScheme.primary
+                )
+            ) {
+                Text(text = stringResource(id = R.string.change_password))
+            }
+        }
+    }
+}
+
+
+@Composable
+fun UpdatePasswordScreen(
+    uiState: RegistrationUiState,
+    snackbarHostState: SnackbarHostState,
+    getBackToPreviousScreen: () -> Unit
+) {
+    val context = LocalContext.current
+    Box {
+        when (uiState) {
```

Comment:
Make a composable named `UpdatePasswordContent` and show main UI in that composable, and call it here. 
Please refer my recent closed PR, you will get to know what I mean.